src/data/fin/interval.lean

```diff
@@ -20,6 +20,14 @@ variables (n : ℕ)
 
 instance : locally_finite_order (fin n) := subtype.locally_finite_order _
 
+instance : Π n, locally_finite_order_top (fin n)
+| 0 := infer_instance
+| (n + 1) := infer_instance
```

Comment:
Are these the only instances?

Reply:
Only instances of what? of `locally_finite_order_top` which are not already implied by other instances? I think so, yes.

Reply:
Can you think of any other "base case" instances? While I'm happy introducing this typeclass for `fin n` (and things built from it) alone, it's hard to evaluate if the design is right with only one real example.

Reply:
Have you seen #14448? The amount of golf alone justifies the design IMO. (I have more changes in local as well)

Reply:
No, I haven't; but #14448 only justifies the ends not the means. I'm looking for evidence that a typeclass of this form is effective for things other than `fin` (or arguments that such things are never going to come up)

Reply:
Then I guess you're looking for `subtype.locally_finite_order_top` and `prod.locally_finite_order_top`? I already have them in this PR.